java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducer.java

```diff
@@ -230,8 +248,10 @@ private void onPutRecordResult(Message msg) {
         }
         
         private void onMetricsResponse(Message msg) {
-            SettableFuture<List<Metric>> f = getFuture(msg);
-            
+            SettableFutureTracker futureTracker = getFuture(msg);
+            // Cancel the future timeout task if present to relieve memory from the ScheduledThreadPoolExecutor
+            futureTracker.cancelTimeoutTaskIfPresent();
```

Comment:
This can be part of `SettableFutureTracker getFuture(Message msg)`

Reply:
done